packages/grafana-data/src/types/dataFrame.ts

```diff
@@ -162,6 +162,11 @@ export interface FieldState {
    * Useful for assigning color to series by looking up a color in a palette using this index
    */
   seriesIndex?: number;
+
+  /**
+   * Location of this field within the context frames results
+   */
+  origin?: DataFrameFieldIndex;
```

Comment:
Name feels ambiguous, maybe sourceFieldIndex ? And what is this index type? (On mobile now)

Reply:
DataFrameFieldIndex was added by @mckn for the legend support -- I am not 100% convinced it is necessary, but not sure how to fix it yet

Reply:
I added `@internal` flags to it -- lets see if we can make it go away

Reply:
since uPlot operates in seriesIdx, i'm not too sure how we can make this go away without joining the frames much higher and having everything downstream work on the joined frame. probably not realistic or necessary since the joining will only be needed in limited cases.